### PR TITLE
Provider api

### PR DESCRIPTION
**Pull request by greghaynes**

### CODE REVIEW

**Comment by cmoulliard** [comment]
Can you please add an issue with the description about the PR ?

File: pkg/controllers/resources/idpbuilder.cnoe.io_custompackages.yaml [hunk 1]
```diff
@@ -3,7 +3,7 @@ apiVersion: apiextensions.k8s.io/v1
 kind: CustomResourceDefinition
 metadata:
   annotations:
-    controller-gen.kubebuilder.io/version: v0.14.0
+    controller-gen.kubebuilder.io/version: v0.12.0
```
Comment (nimakaviani) [inline pkg/controllers/resources/idpbuilder.cnoe.io_custompackages.yaml:6]: looks like you need to bump the version of your controller gen. are you using an older version of idpbuilder or is there any other reason to have this older version used?

File: pkg/controllers/resources/idpbuilder.cnoe.io_custompackages.yaml [hunk 2]
```diff
@@ -19,19 +19,14 @@ spec:
...
         properties:
           apiVersion:
-            description: |-
-              APIVersion defines the versioned schema of this representation of an object.
-              Servers should convert recognized schemas to the latest internal value, and
-              may reject unrecognized values.
-              More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources
+            description: 'APIVersion defines the versioned schema of this representation
+              of an object. Servers should convert recognized schemas to the latest
+              internal value, and may reject unrecognized values. More info: https://git.k8s.io/community/contributors/devel/sig-architecture/api-conventions.md#resources'
```
Comment (nimakaviani) [inline pkg/controllers/resources/idpbuilder.cnoe.io_custompackages.yaml:24]: these diffs will go away if a newer version of controller gen is used

File: pkg/k8s/provider/provider.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+)
+
+type PortMapping struct {
+	HostPort      string
+	ContainerPort string
+}
+
+type KindConfig struct {
+	KindConfigPath string
+}
```
Comment (nimakaviani) [inline pkg/k8s/provider/provider.go:16]: better to move the KindConfig definition under `pkg/k8s/providers/kind` for it to be self contained

**Review by nimakaviani — CHANGES_REQUESTED** [review]

**Comment by nabuskey** [comment]
Closing this due to inactivity. Feel free to reopen.

**Pull request closed without merging**